### PR TITLE
[HIVEMALL-84] Add Docker Support

### PR DESCRIPTION
**Pull request by amaya382**
## What changes were proposed in this pull request?

Add Docker support for development and testing, including build tools for hivemall. Currently, only manual build is supported.

## What type of PR is it?

Feature, Documentation

## What is the Jira issue?

[HIVEMALL-84](https://issues.apache.org/jira/browse/HIVEMALL-84)

## How was this patch tested?

manual tests and automated build tests on DockerHub

## How to use this feature?

See [Document](https://raw.githubusercontent.com/apache/incubator-hivemall/9a3bc5c9aa5b6d26a4c93ba0140c367dc9431090/docs/gitbook/docker/getting_started.md)

~~Official automated build is future work but Dockerfile for automated build is ready. [Trial repository](https://hub.docker.com/r/amaya382/hivemall/)~~
**Known issue: To use automated build, we have to move Dockerfile into root directory**

### CODE REVIEW

**Comment by myui** [comment]
Thanks! I'll look into it.

**Comment by myui** [comment]
@amaya382 Download from cloudera repo failed. 

I'm considering to remove dependencies to CDH soon in https://issues.apache.org/jira/browse/HIVEMALL-86

Could you modify this PR then?

```
Get:180 http://archive.cloudera.com/cdh5/debian/jessie/amd64/cdh/ jessie-cdh5/contrib hive all 1.1.0+cdh5.10.1+906-1.cdh5.10.1.p0.8~jessie-cdh5.10.1 [51.0 MB]
Fetched 365 MB in 35min 49s (170 kB/s)
E: Failed to fetch http://deb.debian.org/debian/pool/main/z/zookeeper/zookeeper_3.4.5+dfsg-2+deb8u1_all.deb  503  first byte timeout [IP: 151.101.0.204 80]

E: Unable to fetch some archives, maybe run apt-get update or try with --fix-missing?
```

File: docs/gitbook/docker/getting_started.md [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+-->
+
+# Getting Started
+
+## How to run
+1. Install the following
+  * `Docker Engine 1.6+`
+  * [OPT] `Docker Compose 1.10+` (optional but recommended)
+2. Build Image
+  * `docker-compose build`
```
Comment (myui) [inline docs/gitbook/docker/getting_started.md]: * `docker build`
* Or, `docker-compose build`

**Comment by amaya382** [comment]
@myui 👌

File: .dockerignore [hunk 2]
```diff
@@ -0,0 +1,6 @@
+.dockerignore
+Dockerfile
+docker-compose.yml
```
Comment (myui) [inline .dockerignore]: Top-level files should be minimum. Move docker related files to `resources/docker`?

**Comment by myui** [comment]
@amaya382 

Could you fix the above conflict by rebase?

Also, changed hadoop dependencies from cdh3 to 2.4.0. Cloudera repository is no more required.
https://github.com/apache/incubator-hivemall/pull/72

File: docs/gitbook/getting_started/installation.md [hunk 3]
```diff
@@ -41,4 +41,14 @@ This automatically loads all Hivemall functions every time you start a Hive sess
...
 source /tmp/define-all.hive;
-```
\ No newline at end of file
+```
+
+
+Other choices
+=============
+
+You can also run Hivemall on the following
```
Comment (myui) [inline docs/gitbook/getting_started/installation.md]: You can also run Hivemall on the following platforms:

**Comment by myui** [comment]
mention me when it's finished.

**Comment by amaya382** [comment]
@myui expecting in a few days

**Comment by amaya382** [comment]
@myui plz check

**Comment by amaya382** [comment]
Notice: Current tree can **NOT** be applied to automated build, we have to move Dockerfile into root directory by [DockerHub limitation](https://github.com/docker/hub-feedback/issues/292)

**Comment by myui** [comment]
@amaya382 Thanks. I'll try this PR in this week.

**Comment by myui** [comment]
@amaya382 merged w/ some modifications. Please check it. Great work!

Tested only w/ Docker engine on Mac, not for docker compose.

**Comment by amaya382** [comment]
@myui trivial errors on doc,
* missing `.` in docker build section
* we have to build hivemall when build image by docker-compose (by build arg `PREBUILD`)

**Comment by myui** [comment]
@amaya382 thanks. could you send pr HIVEMALL-84-2?

**Comment by myui** [comment]
link to jira issue 84.

**Comment by amaya382** [comment]
@myui okay

**Pull request closed without merging**